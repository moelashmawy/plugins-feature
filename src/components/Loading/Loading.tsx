type Props = { width?: number, height?: number}

export const Loading = ({ width = 15, height = 15}: Props) => {
    return <span className="loader" style={{ width: width, height: height }}/>
}