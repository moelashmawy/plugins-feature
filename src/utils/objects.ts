export const getKeyByPropName = (
  data: any,
  keyToFind: string,
  prop: string
): string => {
  const tabName = Object.keys(data).find(
    (tabName) => data[tabName][prop] === keyToFind
  )!;
  return tabName;
};
