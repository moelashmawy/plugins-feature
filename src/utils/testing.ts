import { ReactElement } from 'react';
import { render } from '@testing-library/react';
import { expect } from 'vitest'
import { configureStore } from '@reduxjs/toolkit';
import pluginsSLice from '../state/slices/pluginsSLice';

export const createSnapshot = (ComponentToTest: ReactElement) => {
    const { asFragment } = render(ComponentToTest);
    expect(asFragment()).toMatchSnapshot();
}

export const createTestStore = () => {
  const store = configureStore(
    {
      reducer: {
        plugins: pluginsSLice
      }
    }
  );
  return store;
}