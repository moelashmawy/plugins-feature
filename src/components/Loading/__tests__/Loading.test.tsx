import { it } from 'vitest'
import { createSnapshot } from '../../../utils/testing';
import { Loading } from '../Loading';

it('should render with default width and height', () => {
  createSnapshot(<Loading />);
});

it('should render with width and height 25', () => {
  createSnapshot(<Loading width={25} height={25}/>);
});