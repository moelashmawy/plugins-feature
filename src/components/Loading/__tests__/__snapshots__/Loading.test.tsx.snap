// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render with default width and height 1`] = `
<DocumentFragment>
  <span
    class="loader"
    style="width: 15px; height: 15px;"
  />
</DocumentFragment>
`;

exports[`should render with width and height 25 1`] = `
<DocumentFragment>
  <span
    class="loader"
    style="width: 25px; height: 25px;"
  />
</DocumentFragment>
`;
